src/overlays/actors/ovl_Bg_Hakugin_Post/z_bg_hakugin_post.c

```diff
@@ -21,103 +54,994 @@ const ActorInit Bg_Hakugin_Post_InitVars = {
+    for (i = 0; i < unkStruct->count; i++) {
+        if ((unkStruct->unk_0000[i].unk_34 == 5) || (unkStruct->unk_0000[i].unk_34 == 4)) {
+            unkStruct->unk_0000[i].unk_24 = 0.0f;
+        } else {
+            unkStruct->unk_0000[i].unk_24 -= 2.8f;
+            unkStruct->unk_0000[i].unk_24 *= 0.9f;
+        }
+
+        if ((unkStruct->unk_0000[i].unk_34 == 2) || (unkStruct->unk_0000[i].unk_34 == 3)) {
+            unkStruct->unk_0000[i].unk_20 = 25.0f;
+        } else {
+            unkStruct->unk_0000[i].unk_20 = 0.0f;
+        }
+
+        unkStruct->unk_0000[i].unk_14.x += Math_SinS(unkStruct->unk_0000[i].unk_28) * unkStruct->unk_0000[i].unk_20;
+        unkStruct->unk_0000[i].unk_14.z = (Math_CosS(unkStruct->unk_0000[i].unk_28) * unkStruct->unk_0000[i].unk_20) +
+                                          unkStruct->unk_0000[i].unk_14.z;
+        unkStruct->unk_0000[i].unk_14.y += unkStruct->unk_0000[i].unk_24;
+
+        if ((unkStruct->unk_0000[i].unk_34 == 1) || (unkStruct->unk_0000[i].unk_34 == 2)) {
+            for (j = i - 1; j >= 0; j--) {
+                if ((unkStruct->unk_0000[j].unk_34 == 1) || (unkStruct->unk_0000[j].unk_34 == 2)) {
+                    break;
+                }
+            }
+
+            if (j < 0) {
+                if (unkStruct->unk_0000[i].unk_14.y < 0.0f) {
+                    unkStruct->unk_0000[i].unk_14.y = 0.0f;
+                    unkStruct->unk_0000[i].unk_24 = 0.0f;
+                    if (unkStruct->unk_0000[i].unk_2F & 1) {
+                        unkStruct->unk_0000[i].unk_2F &= ~2;
+                    } else {
+                        unkStruct->unk_0000[i].unk_2F |= 3;
+                    }
+                } else {
+                    unkStruct->unk_0000[i].unk_2F &= ~3;
+                }
+            } else {
+                if (unkStruct->unk_0000[i].unk_14.y <
+                    (unkStruct->unk_0000[j].unk_14.y + D_80A9D880[unkStruct->unk_0000[j].unk_00].unk_00)) {
+                    unkStruct->unk_0000[i].unk_14.y =
+                        unkStruct->unk_0000[j].unk_14.y + D_80A9D880[unkStruct->unk_0000[j].unk_00].unk_00;
+                    unkStruct->unk_0000[i].unk_24 = 0.0f;
+                    if ((unkStruct->unk_0000[i].unk_2F & 1) != 0) {
+                        unkStruct->unk_0000[i].unk_2F &= ~2;
+                    } else {
+                        unkStruct->unk_0000[i].unk_2F |= 3;
+                    }
+                } else {
+                    unkStruct->unk_0000[i].unk_2F &= ~3;
+                }
+            }
+        }
+
+        if (unkStruct->unk_0000[i].unk_34 == 2) {
+            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > 193600.02f) {
+                unkStruct->unk_0000[i].unk_34 = 3;
+            }
+        } else if (unkStruct->unk_0000[i].unk_34 == 3) {
+            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > 278784.03f) {
```

Comment:
```suggestion
            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > SQ(440.0000227272f)) {
                unkStruct->unk_0000[i].unk_34 = 3;
            }
        } else if (unkStruct->unk_0000[i].unk_34 == 3) {
            if (Math3D_XZLengthSquared(unkStruct->unk_0000[i].unk_14.x, unkStruct->unk_0000[i].unk_14.z) > SQ(528.00002840909f)) {
```
Would these squares work? Alternatively would `(SQ(440.0f) + 0.02f)` work to account for the noise

Reply:
Nop, doesn't match.